modules/xmpp/JingleSessionPC.js

```diff
@@ -1799,6 +1907,113 @@ export default class JingleSessionPC extends JingleSession {
     }
 
     /**
+     * Will put and execute on the queue a session modify task. Currently it
+     * only checks the senders attribute of the video content in order to figure
+     * out if the remote peer has video in the inactive state (stored locally
+     * in {@link _remoteVideoActive} - see field description for more info).
+     * @param {jQuery} jingleContents jQuery selector pointing to the jingle
+     * element of the session modify IQ.
+     * @see {@link _remoteVideoActive}
+     * @see {@link _localVideoActive}
+     */
+    modifyContents(jingleContents) {
+        const workFunction = finishedCallback => {
+            if (this._assertNotEnded('modify contents')) {
+                if (this._parseContentSenders(jingleContents)) {
+                    // Will do the sRD/sLD cycle to update SDPs and adjust
+                    // the media direction
+                    this._renegotiate()
+                        .then(finishedCallback, finishedCallback /* (error) */);
+                } else {
+                    finishedCallback();
+                }
+            } else {
+                finishedCallback();
+            }
+        };
+
+        logger.debug('Queued modify contents task');
+
+        this.modificationQueue.push(
+            workFunction,
+            error => {
+                if (error) {
+                    logger.error('modifyContents failed', error);
+                }
+            });
+    }
+
+    /**
+     *
+     * @param jingleContents
+     * @param finishedCallback
+     * @return {boolean}
+     * @private
+     */
+    _parseContentSenders(jingleContents) {
+        const isRemoteVideoActive
+            = this._parseVideoSenders(jingleContents);
+
+        if (isRemoteVideoActive !== null
+            && isRemoteVideoActive !== this._remoteVideoActive) {
+            logger.debug(
+                `${this} new remote video active: ${isRemoteVideoActive}`);
+            this._remoteVideoActive = isRemoteVideoActive;
+        }
+
+        return this.peerconnection.setVideoTransferActive(
+            this._localVideoActive && this._remoteVideoActive);
+    }
+
+    /**
+     * Parses 'senders' attribute of the video content and figures out if remote
+     * peer is interested in receiving our video.
+     * @param {jQuery} jingleContents
+     * @return {boolean|null} <tt>true</tt> if the remote peer is interested in
+     * receiving our video or <tt>false</tt> otherwise (which means that we
+     * should stop sending video). <tt>null</tt> is returned when it failed to
+     * parse the senders attribute of Jingle video contents.
+     * @private
+     */
+    _parseVideoSenders(jingleContents) {
```

Comment:
it would sound like a getter for 'remoteVideoActive', but it's not what it does. It parses Jingle 'contents' element to figure out what information is there